fe2-android/docs/README.md

```diff
@@ -179,44 +170,40 @@ The communication stack within the PoP project is made
 of [multiple layers](https://docs.google.com/document/d/1AeV7JX_SJ30mu9PIwmz24UkIi3jCo6NSYA0sPxdbscU)
 and you need to be familiar with them to understand how communication happens.
 
-The [repository module](https://github.com/dedis/popstellar/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/repository)
-contains some of the network communication logic, it is organized as follows:
-
-* The `remote` module deals with the WebSocket-based connections, using
-  the [Scarlet](https://github.com/Tinder/Scarlet) library.
-* The `LAORepository` exposes the [JSON-RPC](https://www.jsonrpc.org/specification) -based API that
-  provides the publish/subscribe communication abstraction. In it, you will find functions to
-  publish a message, subscribe and unsubscribe to a channel, and so on.
+The [remote module](https://github.com/dedis/popstellar/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote)
+contains most of the network communication logic, it is organized as follows:\
+The `GlobalNetworkManager` is responsible for holding, creating and destroying WebSocket connections.\
+It uses the `ConnectionFactory` to produce new `Connection` objects using
+the [Scarlet](https://github.com/Tinder/Scarlet) library.
 
-The application-level message generation logic is implemented in the `ViewModel` classes, for
-example the request for a LAO creation is in `HomeViewModel`. In the future an API could be created
-to simplify the logic in the application code and reduce repetition.
+The `Connection` is passed to a `LAOConnectionManager` that is responsible for handling the JsonRPC layer of the
```

Comment:
technically it handles also WebSockets, through Scarlet, no? we might want to mention that, since it shows up in the diagram.

Reply:
From the `LAOConnectionManager` point of view, there is nothing to do with the actual websocket it only sees a `Connection` that exposes an input message stream, input event stream and a way to send messages.

I will apply the name changes that you spotted and it should make things clearer.